standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/jdbc/MultiDataSourceJdbcResource.java

```diff
@@ -191,32 +191,23 @@ public Integer execute(ParameterizedCommand command) throws MetaException {
    * call using the query string obtained from {@link ParameterizedBatchCommand#getParameterizedQueryString(DatabaseProduct)},
    * the parameters obtained from {@link ParameterizedBatchCommand#getQueryParameters()}, and the
    * {@link org.springframework.jdbc.core.PreparedStatementSetter} obtained from 
-   * {@link ParameterizedBatchCommand#getPreparedStatementSetter()} methods. The batchSize is coming fomr the 
-   * {@link Configuration} object. After the execution, this method validates the resulted number of affected rows using the
-   * {@link ParameterizedBatchCommand#resultPolicy()} function for each element in the batch.
+   * {@link ParameterizedBatchCommand#getPreparedStatementSetter()} methods. The batchSize is coming from the 
+   * {@link Configuration} object.
    *
    * @param command The {@link ParameterizedBatchCommand} to execute.
-   * @return Returns an integer array,containing the number of affected rows for each element in the batch.
    */
-  public <T> int execute(ParameterizedBatchCommand<T> command) throws MetaException {
+  public <T> void execute(ParameterizedBatchCommand<T> command) throws MetaException {
```

Comment:
In Sql when you execute something it is a convention to return 0 if everything is fine.

Reply:
Originally the return value was the total number of affected rows. The issue here that in case of batch execution the PostgreSQL and MySQL drivers are simply returning -2 as the number of affected rows (which means success BTW), and this makes it impossible sum the values. Since errors are indicated via exceptions inside this method, it would never return anything else than 0.

Reply:
can we keep the jdbc signature?

Reply:
@deniskuzZ you mean modify it to return `int[][]`? The batch call returns the values per batch/per command, not summarized as I did in the old version.

Reply:
yes, int[][]